cfgov/ask_cfpb/scripts/migrate_knowledgebase.py

```diff
@@ -35,6 +36,57 @@
 }
 
 
+def unwrap_soup(soup):
+    soup.html.unwrap()
+    soup.body.unwrap()
+    return unicode(soup)
+
+
+def convert_divs_to_asides(answer_text):
+    soup = bs(answer_text, 'lxml')
+    for div in soup.findAll('div', {'class': 'answer-module'}):
+        div.name = 'aside'
+    return unwrap_soup(soup)
+
+
+def clean_orphan_tips(answer_text):
+    headings = ['TIP', 'WARNING', 'NOTE']
+    soup = bs(answer_text.replace('<br/><br/>', '</p><p>'), 'lxml')
+    h4s = soup('h4')
+    for hed in h4s:
+        if hed.parent.name == 'p' and hed.text.strip(':').upper in headings:
+            wrapper = hed.parent.wrap(
+                soup.new_tag('aside'))
+            wrapper['class'] = 'answer-module'
+    strongs = soup('strong')
+    for strong in strongs:
+        # clean out spans
+        if strong('span'):
+            for span in strong('span'):
+                span.unwrap()
+        # turn strong tips into h4 and wrap in aside
+        if strong.text.strip(':').upper() in headings:
+            strong.name = 'h4'
+            if strong.parent.name == 'p':
+                wrapper = strong.parent.wrap(
+                    soup.new_tag('aside'))
+                wrapper['class'] = 'answer-module'
+                if ':' in wrapper.contents:
+                    i = wrapper.contents.index(':')
+                    del(wrapper.contents[i])
+    return unwrap_soup(soup)
+
+
+def fix_tips(answer_text):
+    try:
+        clean1 = convert_divs_to_asides(answer_text)
+        clean2 = clean_orphan_tips(clean1)
+    except Exception:
```

Comment:
Does that mean that there are cases where the tips cause an exception and get through as part of the answer text? Is that acceptable for the purposes of the migration?